### PR TITLE
Added additional roles to SecurityConfig

### PR DESCRIPTION
**Pull request by ivanke1**
Added ROLE_DRIVER and ROLE_RIDER roles

### CODE REVIEW

**Comment by pconrad** [comment]
This looks good to me and I'm happy to review and merge it.   

But please do a team code review first.

**Review by dingfan123 — DISMISSED** [review]
LGTM

File: src/main/java/edu/ucsb/cs156/gauchoride/config/SecurityConfig.java [hunk 1]
```diff
@@ -88,6 +88,13 @@ private GrantedAuthoritiesMapper userAuthoritiesMapper() {
           if (email.endsWith("@ucsb.edu")) {
             mappedAuthorities.add(new SimpleGrantedAuthority("ROLE_MEMBER"));
           }
+          
+          Optional<User> u = userRepository.findByEmail(email);
+          if (u.isPresent() && u.get().getDriver()) {
+            mappedAuthorities.add(new SimpleGrantedAuthority("ROLE_DRIVER"));
+          } else {
+            mappedAuthorities.add(new SimpleGrantedAuthority("ROLE_RIDER"));
+          }
```
Comment (andrewpengucsb) [inline src/main/java/edu/ucsb/cs156/gauchoride/config/SecurityConfig.java]: Rider role should not be given by default. It needs to be given by an admin with an additional boolean field in the user table that specifies the rider status, like driver, at which point the user can be granted ROLE_RIDER

**Review by andrewpengucsb — DISMISSED** [review]

File: src/main/java/edu/ucsb/cs156/gauchoride/config/SecurityConfig.java (hunk 1 above)
Comment (ivanke1) [reply src/main/java/edu/ucsb/cs156/gauchoride/config/SecurityConfig.java]: FIxed in https://github.com/ucsb-cs156-s23/proj-gauchoride-s23-5pm-3/commit/b2b81791732a07bf4169aa6b7c91631264c13e6a

Comment (ivanke1) [reply src/main/java/edu/ucsb/cs156/gauchoride/config/SecurityConfig.java]: awaiting #31 to fix X's

**Review by Anthonypella — DISMISSED** [review]
LGTM! Logic makes sense

**Review by pconrad — DISMISSED** [review]
LGTM!

**Pull request merged**